### ISSUE TITLE
On Docker using Clickhouse latest e155f535964c causing permission error

### ISSUE THREAD

**Issue by vaguely-tagged** [ClickHouse/ClickHouse#68747]
**Company or project name**
Personal
Put your company name or project description here
Building a docker image from latest
**I have tried the following solutions**: https://clickhouse.com/docs/en/faq/troubleshooting/#troubleshooting-installation-errors
Swapping docker image to 24.3.7 fixes this
**Installation type**
docker on amd64
Packages, docker, single binary, curl?
docker
**Source of the ClickHouse**
dockerhub
A link to the source. Or the command you've tried

**Expected result**
completed install

**The actual result**
Logs showing "get_mempolicy operation not permitted" from clickhouse containter
What you get

**How to reproduce**
boot a clickhouse docker image from latest
* For Linux-based operating systems: provide a script for clear docker container from the official image

* For anything else: steps to reproduce on as much as possible clear system

**Comment by nikitamikhaylov** [comment]
According to the DockerHub `e155f535964c` is a manifest digest of an image with tag `24.8.2.3`  https://hub.docker.com/layers/clickhouse/clickhouse-server/24.8.2.3/images/sha256-e155f535964c4341de3f3297d562fba31d40f867d1837b5469bff0544da323f2

The index digest is `sha256:7f42779cf075896a2014ef90bf092587cce646efd69882b4ebb7cc6299cf7cfe` and it is working perfectly on both Linux VM and Mac. 

```
❯ docker run --platform=linux/amd64 -it clickhouse/clickhouse-server:24.8.2.3
Unable to find image 'clickhouse/clickhouse-server:24.8.2.3' locally
24.8.2.3: Pulling from clickhouse/clickhouse-server
9ea8908f4765: Pull complete
ec1ece6841ae: Pull complete
4f4fb700ef54: Pull complete
20571b4532bb: Pull complete
5a690293f8d3: Pull complete
639ced34cb8a: Pull complete
3587b92dcddb: Pull complete
69b4ffe306d3: Pull complete
d7216eff1bca: Pull complete
Digest: sha256:7f42779cf075896a2014ef90bf092587cce646efd69882b4ebb7cc6299cf7cfe
Status: Downloaded newer image for clickhouse/clickhouse-server:24.8.2.3
WARNING: image with reference clickhouse/clickhouse-server was found but does not match the specified platform: wanted linux/amd64, actual: linux/arm64
Processing configuration file '/etc/clickhouse-server/config.xml'.
Merging configuration file '/etc/clickhouse-server/config.d/docker_related_config.xml'.
Logging trace to /var/log/clickhouse-server/clickhouse-server.log
Logging errors to /var/log/clickhouse-server/clickhouse-server.err.log
2024.08.22 21:14:25.975772 [ 1 ] {} <Information> SentryWriter: Sending crash reports is disabled
2024.08.22 21:14:26.122228 [ 1 ] {} <Information> Application: Starting ClickHouse 24.8.2.3 (revision: 54490, git hash: b54f79ed323f9640c6a01ea227345088f3db1eef, build id: 47C743DBF7D654902DAFAE7F3B176AA5CA6F7A12), PID 1
```

**Comment by spantaleev** [comment]
If you're trying to run ClickHouse as a non-root user, it seems like newer versions require the `CAP_SYS_NICE` capability (`docker run --cap-add=CAP_SYS_NICE ..`).

**Comment by filimonov** [comment]
It seems like it caused by https://github.com/ClickHouse/ClickHouse/pull/67098/

ClickHouse now tries to detect the memory bound to numa nodes, that requires get_mempolicy syscall. 

This syscall is disabled in docker using seccomp security profiles:
https://docs.docker.com/engine/security/seccomp/

CAP_SYS_NICE changes that:

https://github.com/moby/moby/blob/e0384101da0dc812ef4981924d461af94ec4486c/profiles/seccomp/default.json#L786-L797
https://github.com/moby/moby/pull/37242

cc. @antonio2368 

Wondering if some check can be added to ensure that `get_mempolicy` is available? And disable the feature otherwise.

**Comment by antonio2368** [comment]
> Wondering if some check can be added to ensure that get_mempolicy is available? And disable the feature otherwise.

Yeah, we need to add something like that. 
Numa nodes are used only to set correct memory limit so it shouldn't affect CH in negative way if it doesn't work.
I'll check it out

**Comment by filimonov** [comment]
@antonio2368  

this simple fix helps: https://github.com/numactl/numactl/pull/232 

That requires to do change in numactl submodule, and later update submodule. Probably it will be simpler to you, since you have all permissions in both.

**Comment by ddxv** [comment]
@alexey-sh I also was pulling new versions to get this to go away (often would open logs and see hundreds of the error). Coincidentally, I changed from 23.10 to 23.11 and I just noticed it's gone for me now.

**Comment by ddxv** [comment]
@alexey-sh / an FYI for anyone else, I eventually hit the `get_mempolicy operation not permitted` errors again as the memory filled up. I'm trying to squeeze ClickHouse into a very low memory setting, so I believe that is what is causing it for me, and likely not a bug that needs to have an issue opened. Maybe that helps you though if you're trying to see what is causing it.

**Comment by filimonov** [comment]
> The issue is not fixed.
> 
> `sudo docker run --platform=linux/amd64 -it clickhouse/clickhouse-server:24.10.3.21`

It was fixed in 24.11

**Comment by alexey-sh** [comment]
@filimonov

I have disclosed all the information currently at my disposal. However, should you require further clarification, I would be pleased to elaborate on this particular line of code.

The command `sudo docker run --platform=linux/amd64 -it clickhouse/clickhouse-server:24.10.3.21` initiates a Docker container for the specified version of the ClickHouse server (`24.10.3.21`). The `--platform=linux/amd64` flag ensures that the container emulates the `amd64` architecture, which may be necessary when operating on systems with differing native architectures (such as ARM-based Macs). The `-it` flags enable interactive mode and allocate a pseudo-TTY, allowing for direct user interaction within the container. The use of `sudo` implies that elevated privileges are required to execute the Docker command.

I believe everyone can copy the command from a browser, paste into a terminal and execute it. 

In any case, I have no interest in continuing to debug this software, searching for the root cause, attempting to force a resolution to the error, or proving its existence beyond what I have already provided (command + screenshot of the result).
